### ISSUE TITLE
Make notebook cell focus & indicator themeable

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode#98984]
![image](https://user-images.githubusercontent.com/35271042/83456771-2c234980-a415-11ea-8595-ab1524b0b21b.png)

Ideally we should allow themes to theme the cell focus

**Comment by misolori** [comment]
Added the following color tokens for notebook cell selection & focus:

- `notebook.selectedCellIndicator`
- `notebook.focusedCellIndicator`

![image](https://user-images.githubusercontent.com/35271042/84104767-4b991400-a9cb-11ea-8b24-7b676dcadff6.png)

We may want to change "indicator" to "border to be consistent, went with "indicator" since that's what we're currently using.

**Comment by misolori** [comment]
Also for HC:

![image](https://user-images.githubusercontent.com/35271042/84104897-9155dc80-a9cb-11ea-9229-87a61e1f4e84.png)

**Comment by roblourens** [comment]
We should change the terminology to `focusedEditorIndicator` and `focusedCellIndicator` since this doesn't involve the list's focus/selection concepts, but we will probably have to come up with colors for that difference when we have multiselect.

**Comment by misolori** [comment]
Updated the color tokens to `notebook.focusedCellIndicator` and `notebook.focusedEditorIndicator`.

**Comment by Tyriar** [comment]
I know the theme keys are a bit of an mess but we don't want to make them more inconsistent. We always refer to editor focus colors as borders for example.

**Comment by misolori** [comment]
Verification: test that you can use `notebook.focusedEditorIndicator` to theme the editor focus and `notebook.cellFocusBackground` for the cell focus bg color.

**Comment by isidorn** [comment]
I have verified that these colors do work however:
`focusedEditorIndicator` is a weird name, especialy since we already have a color `focusBorder`. Due to that I suggest to rename this one to `notebook.focusBorder` or `notebook.focusedEditorBorder`

`notebook.cellFocusBackground` is not a good name imho, since you also use this background on hover. And on hover we do not change focus. So either do not apply this color on hover, or rename it.

Due to this reopening this issue.
fyi @aeschli on color names

**Comment by misolori** [comment]
The problem we have is that there are various states:

- Hover (`notebook.cellFocusBackground`)
- Cell Focus (`notebook.cellFocusBackground` and `notebook.focusedCellShadow`)
- Editor Focus (`notebook.focusedEditorIndicator`)

I'm fine with making these consistent and possibly adding a new one for hover. So it would be:

- Hover (`notebook.hoverBackground``)
- Cell Focus (`notebook.focusedCellBackground` and `notebook.focusedCellShadow`)
- Editor Focus (`notebook.focusedEditorBorder`)

Thoughts?

**Comment by isidorn** [comment]
That makes sense for me.
Maybe we should also ping others @roblourens @jrieken @aeschli @rebornix

**Comment by misolori** [comment]
So I'm working on making sure the focus selection on the cell matches our theme colors, but am wondering if this is too much focus orange?

![image](https://user-images.githubusercontent.com/35271042/86277129-0c558180-bb8b-11ea-87e1-0f665ffd7469.png)

**Comment by Tyriar** [comment]
@misolori the current pattern seems to be:

Active, not focused list:

![image](https://user-images.githubusercontent.com/2193314/86278703-afa79600-bb8d-11ea-9c05-4bf2e05946c1.png)

Hovered list:

![image](https://user-images.githubusercontent.com/2193314/86278738-b930fe00-bb8d-11ea-86c7-6b89e4fddbd6.png)

Focused list:

![image](https://user-images.githubusercontent.com/2193314/86278754-c0f0a280-bb8d-11ea-815c-53d4c0a0faf0.png)

In the editor the active line is solid regardless of whether it's focused or not:

![image](https://user-images.githubusercontent.com/2193314/86278860-e54c7f00-bb8d-11ea-92d5-da3f161bd15b.png)

Maybe try a dotted line for the outer focus borders when there is an inner focus?

**Comment by misolori** [comment]
Updated the token colors as:

- Hover (`notebook.hoverBackground`)
- Cell Focus
  - `notebook.focusedCellBorder`
  - `notebook.focusedCellBackground`
  - `notebook.focusedCellShadow`
- Editor Focus (`notebook.focusedEditorBorder`)

Additionally, I've updated this in HC so that when the cell is in focus it has the solid focus border and when the editor is in focus we use the dashed border:

![Kapture 2020-07-01 at 13 40 17](https://user-images.githubusercontent.com/35271042/86289757-17b3a780-bba1-11ea-8073-a48d3f5ee436.gif)

**Comment by isidorn** [comment]
@misolori thanks for looking at this. While I like the changes, there are still a couple of things:
* Description of the colors should be tweaked to be more precise
* `notebook.focusedCellBackground` is still applied on hover
* `notebook.hoverBackground` is not recognized in JSON and does not seem to have any effect
* `focusedCellBorder` is only top and bottom line, so it is not actually a border. This is fine by me, but might be slightly unexpected

Due to that reopening this issue and you can decide if you would prefer to close it and movi this out to a seperate issue. Thanks!

**Comment by misolori** [comment]
Updated the description for all of them.

The cell hover color is taken from `notebook.focusedCellBackground` but @ 50% opacity (without trying to introduce new colors). And the actual color token is `notebook.cellHoverBackground` since we have other hover color tokens and need this one to specify that it's for the cell.

**Comment by bpasero** [comment]
@misolori it looks like the input field is falling out of the cell:

![image](https://user-images.githubusercontent.com/900690/86463880-a43fa080-bd2e-11ea-83f2-48bdfe546831.png)

Please do not forget to update our docs for the new theme colors and release notes.
